intake/catalog/base.py

```diff
@@ -298,26 +302,36 @@ def __init__(self, url, http_args={}, page_size=None, **kwargs):
         page_size : int, optional
             The number of entries fetched at a time during iteration.
             Default is None (no pagination; fetch all entries in bulk).
+        query : Iterable or None
```

Comment:
Yes, client-side search is also valuable. Client-side searching doesn't need to stash and pass through state of "the queries that got us to this point" because it can just materialize the entire resultant (local) Catalog. On a RemoteCatalog, I assume that we need the server (in my application's case, MongoDB) to parse the parameters to `search(...)`, so we have to aggregate them as state carried along with the `RemoteCatalog`.

Reply:
Not sure I understand the question. Let's remember to clarify this one on the call.